src/pyhf/cli/stats.py

```diff
@@ -0,0 +1,67 @@
+import logging
+
+import click
+import json
+
+from ..utils import hypotest, EqDelimStringParamType
+from ..workspace import Workspace
+from .. import tensorlib, set_backend, optimize
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+
+@click.group(name='stats')
+def cli():
+    pass
+
+
+@cli.command()
+@click.argument('workspace', default='-')
+@click.option(
+    '--output-file',
+    help='The location of the output json file. If not specified, prints to screen.',
+    default=None,
+)
+@click.option('--measurement', default=None)
+@click.option('-p', '--patch', multiple=True)
+@click.option('--testpoi', default=1.0)
+@click.option('--teststat', type=click.Choice(['q', 'qtilde']), default='qtilde')
+@click.option('--optimizer')
+@click.option('--optconf', type=EqDelimStringParamType(), multiple=True)
+def cls(
+    workspace, output_file, measurement, patch, testpoi, teststat, optimizer, optconf
+):
+    with click.open_file(workspace, 'r') as specstream:
+        wspec = json.load(specstream)
+
+    w = Workspace(wspec)
+
+    is_qtilde = teststat == 'qtilde'
+
+    patches = [json.loads(click.open_file(pfile, 'r').read()) for pfile in patch]
+    p = w.model(
+        measurement_name=measurement,
+        patches=patches,
+        modifier_settings={
+            'normsys': {'interpcode': 'code4'},
+            'histosys': {'interpcode': 'code4p'},
+        },
+    )
+
+    optconf = {k: v for item in optconf for k, v in item.items()}
+
+    # set the new optimizer
+    if optimizer:
+        new_optimizer = getattr(optimize, optimizer)
+        set_backend(tensorlib, new_optimizer(**optconf))
+
+    result = hypotest(testpoi, w.data(p), p, qtilde=is_qtilde, return_expected_set=True)
```

Comment:
I'd advocate to in general avoid 1 letter variables 
```suggestion
    result = hypotest(testpoi, ws.data(pdf), pdf, qtilde=is_qtilde, return_expected_set=True)
```